osu.Game/Screens/Select/FilterCriteria.cs

```diff
@@ -114,6 +114,27 @@ public string SearchText
 
         public IRulesetFilterCriteria? RulesetCriteria { get; set; }
 
+        public struct OptionalArray<T> : IEquatable<OptionalArray<T>>
+            where T : struct
+        {
+            public bool HasFilter => Values?.Length > 0;
+
+            public bool IsInRange(T value)
+            {
+                return Values?.Contains(value) ?? false;
```

Comment:
Please don't use patterns like `?? false` because it is confusing to read - it's a bit double-negationy.

```suggestion
                return Values?.Contains(value) == true;
```

is miles better readability wise imo.